.travis.yml

```diff
@@ -7,15 +7,15 @@ cache:
   ccache: True
 
 
-# We need a newer version of zlib to build notmuch from source as is available
-# in precise.
-dist: trusty
+# Python 3.7 is not supported in trusty
+dist: xenial
```

Comment:
Since when is that possible? I never saw it in the docs? (But the log claims it works.)

Reply:
I found it in a bug report about python 3.7 support. It's apparently not fully supported yet, but it seems to work. The suggested workaround for python 3.7 support was to build a docker image, which we can do but it's much more involved than just setting "xenial"

Reply:
Can we live with some older libraries than the ones we currently build by hand? Here is what we build by hand and what xenial has:
* gpgme 1.9 -> [1.6](https://packages.ubuntu.com/xenial/libgpgme11)
* libassuan (dep for gpgme) 2.4.3 -> [2.4.2](https://packages.ubuntu.com/xenial/libassuan-dev)
* libgpg-error (dep for gpgme) 1.27 -> [1.21](https://packages.ubuntu.com/xenial/libgpg-error-dev)
* notmuch (we require 0.23 in setup.py) git (xenial is to low but we could switch to a tarball)

Reply:
I think it's worth trying to use the existing versions of stuff (I'm pretty sure that libassuan can be dropped at least). I'll push a few patches and see what happens.